src/cider/nrepl/middleware/complete.clj

```diff
@@ -2,33 +2,46 @@
   (:require
    [cider.nrepl.middleware.util.cljs :as cljs]
    [cider.nrepl.middleware.util.error-handling :refer [with-safe-transport]]
-   [cljs-tooling.complete :as cljs-complete]
-   [compliment.core :as jvm-complete]
-   [compliment.utils :as jvm-complete-utils]
+   [compliment.core :as complete]
+   [compliment.utils :as complete-utils]
    [orchard.misc :as misc]
-   [suitable.complete-for-nrepl :as suitable]))
+   [suitable.complete-for-nrepl :as suitable]
+   [suitable.compliment.sources.cljs :as suitable-sources]))
 
-(defn- cljs-complete
-  [{:keys [enhanced-cljs-completion?] :as msg} cljs-env ns prefix extra-metadata]
-  (concat (cljs-complete/completions cljs-env prefix {:context-ns ns
-                                                      :extra-metadata extra-metadata})
-          (when enhanced-cljs-completion? (suitable/complete-for-nrepl msg))))
+(def +clj-sources+
+  "Source keywords for Clojure completions."
+  [:compliment.sources.special-forms/literals
+   :compliment.sources.class-members/static-members
+   :compliment.sources.ns-mappings/ns-mappings
+   :compliment.sources.resources/resources
+   :compliment.sources.keywords/keywords
+   :compliment.sources.local-bindings/local-bindings
+   :compliment.sources.class-members/members
+   :compliment.sources.namespaces-and-classes/namespaces-and-classes
+   :compliment.sources.special-forms/special-forms])
+
+(def +cljs-sources+
+  "Source keywords for ClojureScript completions."
+  [::suitable-sources/cljs-source])
 
 (defn complete
-  [{:keys [ns symbol context extra-metadata] :as msg}]
-  (let [ns (misc/as-sym ns)
-        prefix (str symbol)
-        extra-metadata (set (map keyword extra-metadata))]
+  [{:keys [ns symbol context extra-metadata enhanced-cljs-completion?] :as msg}]
+  (let [prefix (str symbol)
+        completion-opts {:ns (misc/as-sym ns)
+                         :context context
+                         :extra-metadata (set (map keyword extra-metadata))}]
     (if-let [cljs-env (cljs/grab-cljs-env msg)]
-      (cljs-complete msg cljs-env ns prefix extra-metadata)
-      (jvm-complete/completions prefix {:ns ns
-                                        :context context
-                                        :extra-metadata extra-metadata}))))
+      (binding [suitable-sources/*compiler-env* cljs-env]
+        (concat (complete/completions prefix (merge completion-opts {:sources +cljs-sources+}))
+                (when enhanced-cljs-completion? (suitable/complete-for-nrepl msg))))
```

Comment:
Shouldn't this just add one more compliment source?

Reply:
As far I understand, the enhanced completions are not a `defsource` at the moment, maybe @rksm can expand on that

Reply:
Okay, let's hear what he'll tell us about those. It just seems super strange to me to have one set of completions as compliment source and the others as something else.